api/src/main/java/io/cloudevents/types/Time.java

```diff
@@ -31,16 +34,56 @@ private Time() {
     }
 
     /**
-     * Parse a {@link String} RFC3339 compliant as {@link OffsetDateTime}
+     * Parse a {@link String} RFC3339 compliant as {@link OffsetDateTime}.
+     *
+     * @param time the value to parse as time
+     * @return the parsed {@link OffsetDateTime}
+     * @throws DateTimeParseException if something went wrong when parsing the provided time.
      */
     public static OffsetDateTime parseTime(String time) throws DateTimeParseException {
         return OffsetDateTime.parse(time);
     }
 
     /**
-     * Convert a {@link OffsetDateTime} to a RFC3339 compliant {@link String}
+     * Parse an attribute/extension with RFC3339 compliant {@link String} value as {@link OffsetDateTime}.
+     *
+     * @param attributeName the attribute/extension name
+     * @param time          the value to parse as time
+     * @return the parsed {@link OffsetDateTime}
+     * @throws CloudEventRWException if something went wrong when parsing the attribute/extension.
+     */
+    public static OffsetDateTime parseTime(String attributeName, String time) throws CloudEventRWException {
+        try {
+            return parseTime(time);
+        } catch (DateTimeParseException e) {
+            throw CloudEventRWException.newInvalidAttributeValue(attributeName, time, e);
+        }
+    }
+
+    /**
+     * Convert a {@link OffsetDateTime} to a RFC3339 compliant {@link String}.
+     *
+     * @param time the time to write as {@link String}
+     * @return the serialized time
+     * @throws DateTimeException if something went wrong when serializing the provided time.
      */
-    public static String writeTime(OffsetDateTime time) throws DateTimeParseException {
+    public static String writeTime(OffsetDateTime time) throws DateTimeException {
         return ISO_OFFSET_DATE_TIME.format(time);
     }
+
+    /**
+     * Convert an attribute/extension {@link OffsetDateTime} to a RFC3339 compliant {@link String}.
+     *
+     * @param attributeName the attribute/extension name
+     * @param time          the time to write as {@link String}
+     * @return the serialized time
+     * @throws CloudEventRWException if something went wrong when serializing the attribute/extension.
+     */
+    public static String writeTime(String attributeName, OffsetDateTime time) throws DateTimeException {
```

Comment:
When I first saw call to this method I was confused why it needs attributeName. I found unusual that `writeTime()` needs that. Maybe keep explicit exception catching?

Reply:
So you propose to remove these overrides?

Reply:
At first I thought we should remove the overload because the overload is used only on one place. But now I see that that there is similar `parseTime()` method which is used at multiple places. So better keep it as it is.